### DISCUSSION TITLE
Does scaledjob supports sidecar containers?

### DISCUSSION THREAD

**Discussion by chaithraPadmar** [kedacore/keda#6044] [Q&A / Need Help]
Team,

We would like to implement sidecar containers for the app where we are using scaledjobs, 

Could you please give some sample ref for scaleejob with multicontainer

The second container is to ship logs to openserach

Reagards
Chaithra

**Comment by JorTurFer** [comment]
Hello
It's the same as using a single container. You have to set your containers spec as part of the job reference

**Comment by chaithraPadmar** [comment]
Thank you for your email

**Comment by chaithraPadmar** [comment]
Hi

Anothr query is , while using 2.13.0 keda version..

Can we use trigger auth with  secrets?

Kubernetes version in use is 1.29

**Reply by JorTurFer** [reply]
What do you mean? For the scaler? yes you can. You can use TriggerAuthentication reading k8s secrets

**Comment by chaithraPadmar** [comment]
Thanks

**Comment by chaithraPadmar** [comment]
Hi team,
I have 3 queries.. kindly help

1. For scaled job if we keep rollout strategy as gradual and post
deployment if we manualy clean up spinned pods, why the newly spinned pods
donot pick new configs?

Consider minreplica 5, maxreplica 100
Job queue is 3 as example.


2. What is the hard limit which stops perpool 250 jobs/pods in
azure-pipelines trigger while using scaledjobs?

Can we modify it.. does the modification requires restart?


3. Can we just restart scaledjob without running the whole helm chart
redeployment?

**Comment by chaithraPadmar** [comment]
Hi team,
I have 3 queries.. kindly help

1. For scaled job if we keep rollout strategy as gradual and post
deployment if we manualy clean up spinned pods, why the newly spinned pods
donot pick new configs?

Consider minreplica 5, maxreplica 100
Job queue is 3 as example.


2. What is the hard limit which stops perpool 250 jobs/pods in
azure-pipelines trigger while using scaledjobs?

Can we modify it.. does the modification requires restart?


3. Can we just restart scaledjob without running the whole helm chart
redeployment?.. we are deploying scaled jobs as part of helm with other
dependent resources

Kindly help with these queries..
We are using scaled job on aws eks since 6 months on  our prod


Regards
Chaithra

**Comment by JorTurFer** [comment]
Hello

1. Are you deleting the job or the pod?
2. What do you mean? KEDA doesn't have a hard limit, you can put 2000 as max replicas if you want. The limit is your licenses
3. What do you mean as restart? You can modify the ScaledJob when you want

**Comment by chaithraPadmar** [comment]
hi,

1. Am deleting pods; do i need to delete jobs and test?

2. we have parallel licenses 500+; case that am facing is if i keep
minreplica as 20 and maxreplica as 350; eventhough 400 jobs in queue
constantly to a pool; the running jobs on azure pipeline GUI  wont cross
250;
but if i increase minReplica equal to maxReplica ; say 350; then i can see
around 300 (assume that 50 will be in complete/initiation state for app)
what is *jobToFetch* variable on scaledjob?

3. In the above scenario trying to set a different value to jobToFetch; it
is not reflecting.

Regards
Chaithra


On Thu, Aug 29, 2024 at 2:23 AM Jorge Turrado Ferrero <
***@***.***> wrote:

> Hello
>
>    1. Are you deleting the job or the pod?
>    2. What do you mean? KEDA doesn't have a hard limit, you can put 2000
>    as max replicas if you want. The limit is your licenses
>    3. What do you mean as restart? You can modify the ScaledJob when you
>    want
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/kedacore/keda/discussions/6044#discussioncomment-10479808>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AVVEPIK7GIPXJOXYQPTKGPDZTY2GPAVCNFSM6AAAAABMDW46E6VHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTANBXHE4DAOA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Thank's and  Regard's
Chaithra

**Comment by JorTurFer** [comment]
Hello

1. If you kill the pod but the job is still there, the job hasn't changed. If you want to force the rollout of the new version, you have to delete the jobs
2. I don't have any info about a limitation here tbh. We limit the jobs to `jobsToFetch` to reduce the pulled jobs and improve the performance, but maybe you need to increase to 500-600 or even more.
3. Upgrading the value should be enough to automatically update the value internally.

What KEDA version are you using?

**Comment by chaithraPadmar** [comment]
Hi

Thanks for clarifying things

We are currently using 1.25 kubernetes and 2.10.0 keda

In process of migrating to 1.29 kubernetes and 2.13.0 keda

Regards
Chaithra

**Comment by chaithraPadmar** [comment]
Hi team,

What is the best scaling strategy while using azure-pipelines trigger
amongst
Default
Custom
Accurate

I do have a problem, where am missing something which sets maxscale to 220
When am using default scaling strategy

If i use accurate.. it runs max replica always

.. donot have a example for trigger azure pipelines with custom startegy
and values

Please help
Regards
Chaithra

**Comment by chaithraPadmar** [comment]
Hi,


https://github.com/kedacore/keda/blob/release/v2.10/pkg/scalers/azure_pipelines_scaler.go#L10

Here line 334 restricts first 200 jobs?
Regards
Chaithra

**Comment by chaithraPadmar** [comment]
https://github.com/kedacore/keda/discussions/6146

Opened news issue/discussion to close this thread
